packages/core-sdk/resolvers/account-resolvers/interface/query.graphql

```diff
@@ -1,72 +1,130 @@
 #import { Token } into TokenResolver from "ens/interface.token-resolvers.defiwrapper.eth"
 
 type Query {
-  getTokenBalances(address: String!): [TokenBalance!]!
+  getTokenBalances(accountAddress: String!): TokenBalancesList!
 
-  # TODO
-  # getTokenTransfers(address: String!): TransfersList!
+  getTokenTransfers(
+    accountAddress: String!
+    tokenAddress: String!
+    vsCurrency: String!
+    options: Options
+  ): TransfersList!
 
-  # TODO
-  # getTransactions(address: String!): TransfersList!
+  getTransactions(accountAddress: String!, vsCurrency: String!, options: Options): TransactionsList!
+}
+
+type TokenBalancesList {
+  account: String!
+  chainId: String!
+  tokenBalances: [TokenBalance!]!
 }
 
 type TokenBalance {
   token: TokenResolver_Token!
   balance: String!
 }
 
-# TODO
-# enum TransferType {
-#   IN,
-#   OUT,
-# }
-
-# type Transfer {
-#   txHash: String!
-#   from: String!
-#   to: String!
-#   value: String!  # BigDecimal
-#   transferType: TransferType!
-#   timestamp: String!
-# }
-
-# type TransfersList {
-#   token: Token!
-#   transfers: [Transfer!]!
-#   pagination: Pagination!
-# }
-
-# type Options {
-#   pagination: PaginationOptions
-#   blockRange: BlockRangeOptions
-# }
-
-# type PaginationOptions {
-#   page: Int!
-#   perPage: Int!
-# }
-
-# type BlockRangeOptions {
-#   startBlock: Int  # default: 0
-#   endBlock: Int  # default: latest
-# }
-
-# type Pagination {
-#   total: Int!
-#   perPage: Int!
-#   page: Int!
-#   hasMore: Boolean!
-# }
-
-# type Transaction {
-#   hash: String!
-#   from: String!
-#   to: String!
-#   value: String!  # BigDecimal
-#   gasUsed: Int!  # Gas used for the transaction
-#   gasLimit: Int!  # Gas offered for the transaction
-#   gasPrice: String!  # Gas price in gwei (BigDecimal)
-#   timestamp: String!  # Timestamp of the transaction
-#   blockHeight: Int!  # Block height of the transaction
-#   offset: Int!
-# }
+type Options {
+  pagination: PaginationOptions
+  blockRange: BlockRangeOptions
+}
+
+type PaginationOptions {
+  page: Int!
+  perPage: Int!
+}
+
+type BlockRangeOptions {
+  startBlock: Int # default: 0
+  endBlock: Int # default: latest
+}
+
+type Pagination {
+  total: Int
+  perPage: Int
+  page: Int
+  hasMore: Boolean
+}
+
+type GasInfo {
+  offered: String! # Gas offered for the transaction
+  spent: String! # Gas spent for the transaction
+  price: String! # Gas price in wei (BigDecimal)
+  quoteRate: String! # rate of native currency in quoted currency (BigDecimal)
+  quote: String! # Gas price in quote currency (BigDecimal)
+}
+
+type Transaction {
```

Comment:
Should this include a `chainId` prop or is it always in a TransactionList? Also, do we need a `status` prop that indicates whether a transaction succeeded or not? Or will we filter to include only successful transactions?

Reply:
If this is a bad question because it's handled by the API, just let me know. :)

Reply:
Nice catch! I assumed that API would only include successful transactions but I will check with them to know for sure.

Reply:
I checked and found we do so thanks for catching this, I will add a new field